controllers/StockApiController.php

```diff
@@ -322,7 +326,23 @@ public function ConsumeProductByBarcode(\Psr\Http\Message\ServerRequestInterface
 	{
 		try
 		{
-			$args['productId'] = $this->getStockService()->GetProductIdFromBarcode($args['barcode']);
+			if (Grocycode::Validate($barcode))
+			{
+				$gc = new Grocycode($barcode);
+				if ($gc->GetType() != Grocycode::PRODUCT)
+				{
+					throw new \Exception('Invalid grocycode type');
+				}
+				$args['productId'] = $gc->GetId();
+				if ($gc->GetExtraData())
+				{
+					$args['stock_id'] = $gc->GetExtraData()[0];
+				}
+			}
```

Comment:
Exactly that check is already present (you added it) in `StockService->GetProductIdFromBarcode`, why not reuse it?

Reply:
The fix in `StockService` is for all the endpoints that get the product id from the barcode - the `ConsumeProductByBarcode` endpoint needs additional processing to get the stock entry id from the barcode.

While it could be changed to still hit the `StockService` the gain is minimal as we still need to check for a valid product grocycode to extract the ExtraData.